### PR TITLE
New strategy for training

### LINKED ISSUE #28: Change loss accumulation
Try using `torch.sum` instead of `torch.mean` when accumulating the loss for each `batch_size`.

### LINKED ISSUE #29: Mini architecture
Create a new agent class which has much fewer parameters than our current small architecture.

### LINKED ISSUE #31: Next models to be trained
These are the combinations of architectures and training strategies that we want to train next.

Basic reward, small architecture (to see if bad performance is due to our extreme reward system)
Basic reward(?), mini architecture, consider possibility of playing more than 1M episodes with mini architecture
Reward(?), mini architecture, play against minimax agent
Reward(?), small architecture, play against minimax agent
Basic reward, large architecture, play against minimax agent

### Important note
We have introduced other hyperparameters that we want to explore, but our current remaining time for the project is limited.
_Suggestion_: Try out some of our new hyperparameters (for instance #28 and #30) against the model in #29 but maybe just for 100k games. Perhaps this could be done on my pc?

### PR DESCRIPTION
**Pull request by RasmusBrostroem**
Implemented the mini architecture

closes #29 

Tested loss accumulation and decided on loss_mean instead of loss_sum

closes #28 

Added new agents to train and option for pretrain agents. Also, changed the minimax training strategy and changed max_depth to 0

closes #31 

Saved some of the new agents and the neptune id's for the models already trained.

### CODE REVIEW

File: Loss_mean_agents.csv [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+DefenderV1,50,-500,20,-50,20,50000,100,0.0001,0.1,False,0.99,Large,False,CON-83,
+WinnerV2,500,-50,20,-100,100,10000,100,0.0001,0.05,False,0.9,Large,False,CON-84,
+RuleBoi,200,-50,50,-500,50,20000,50,0.0001,0.1,True,0.4,Small,False,CON-89,
+RuleBoiV2,200,-50,50,-500,20,50000,100,0.0001,0.1,True,0.9,Large,False,CON-90,
+AverageJoe,100,-50,35,-100,15,20000,50,0.001,0,False,0.99,Small,False,CON-94,
+AverageJoeV2,100,-50,35,-100,15,10000,50,0.001,0,False,0.99,Small,True,,AverageJoe_gen_14.pth
+BasicBitch,100,-100,50,-50,4,50000,50,0.001,0,False,0.99,Mini,True,,
+BasicBitchV2,100,-100,50,-50,4,50000,50,0.001,0,False,0.99,Small,True,,
+TequilaBoi,50,-100,35,-50,50,20000,50,0.001,0,False,0.99,Mini,False,,
+TequilaBoiV2,50,-100,35,-50,50,20000,50,0.001,0,False,0.99,Small,False,,
```
Comment (jbirkesteen) [inline Loss_mean_agents.csv:12]: Something looks a bit fishy in this file.
Why is it that line 2-7 end with only one comma, line 8 ends with 0 commas, and the following line ends with 2 commas?
I would expect all lines, except line 8, to end with 2 commas, as they have a missing value in the last column.
But pandas reads it correctly? I guess it looks like it does in the Neptune runs at least.

**Review by jbirkesteen — COMMENTED** [review]
Great work!
I have 2 questions before merging.
Is it on purpose that you've requested pull of AverageJoe? 
For the second, see my comment on the new .csv.

**Comment by jbirkesteen** [comment]
I just now realised that you've merged #36.
I may have misunderstood you, but I thought we didn't want to have all of the models in our repo (mostly due to concerns about space)?
Anyhow: We need to either remove AgentParameters from the .gitignore or explicitly remove model parameters in order for them to be ignored.

File: Loss_mean_agents.csv (hunk 1 above)
Comment (jbirkesteen) [reply Loss_mean_agents.csv:12]: I had forgotten Netpune column. Everything looks fine.
